backend/Application/Aggregates/Contract/ContractAggregate.cs

```diff
@@ -325,12 +387,101 @@ await repository
                         transactionIndex,
                         eventIndex,
                         contractModuleDeployed.ModuleRef,
+                        sender,
                         source,
                         blockSlotTime
                     ));
                 break;
         }
     }
+
+    /// <summary>
+    /// Store rejected events if present in block item summary and if related to contract- or modules.
+    ///
+    /// Return 1 if a rejected event was created.
+    /// </summary>
+    private static async Task<uint> StorePossibleRejectEvent(
+        IContractRepository repository, 
+        BlockInfo blockInfo, 
+        AccountTransactionDetails details,
+        BlockItemSummary blockItemSummary
+    )
+    {
+        if (details.Effects is not None none || !IsRelevantReject(none.RejectReason, out var rejectReason))
+        {
+            return 0;
+        }
+        
+        await StoreReject(
+            ImportSource.NodeImport,
+            repository,
+            rejectReason!,
+            AccountAddress.From(details.Sender),
+            blockInfo.BlockHeight,
+            blockInfo.BlockSlotTime,
+            blockItemSummary.TransactionHash.ToString(),
+            blockItemSummary.Index
+        );
+
+        return 1;
+    }
+
+    /// <summary>
+    /// Map relevant rejected reasons to <see cref="TransactionRejectReason"/>.
+    /// </summary>
+    private static bool IsRelevantReject(IRejectReason rejectReason, out TransactionRejectReason? rejected)
```

Comment:
I will refactor such that is starts with `Try..`. Then it aligns with sections further down in the documentation you have linked.
https://learn.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2015/code-quality/ca1021-avoid-out-parameters?view=vs-2015&redirectedfrom=MSDN#try-pattern-methods

The `Try..`pattern is quite known in C# and personally I really like it because you tell the developer, that this method may return a nullable which you need to handle.
When returning nullable the developers either way needs to do the null check and handle different cases.

Hence I would like to stick with what is done after renaming :-)